ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -332,51 +317,62 @@ private void readFullAccountAndUpdateCache(String pathToFullAccountMetadata, boo
           AccountInfoMap newAccountInfoMap = new AccountInfoMap(remoteAccountMap);
-          Map<Short, Account> oldIdToAccountMap = accountInfoMapRef.get().idToAccountMap;
-          accountInfoMapRef.set(newAccountInfoMap);
-          Map<Short, Account> idToUpdatedAccounts = new HashMap<>();
-          for (Account newAccount : newAccountInfoMap.getAccounts()) {
-            if (!newAccount.equals(oldIdToAccountMap.get(newAccount.getId()))) {
-              idToUpdatedAccounts.put(newAccount.getId(), newAccount);
-            }
-          }
-          if (idToUpdatedAccounts.size() > 0) {
-            logger.info("Received updates for {} accounts by listener={}. Account IDs={}", idToUpdatedAccounts.size(),
-                isCalledFromListener, idToUpdatedAccounts.keySet());
-            // @todo In long run, this metric is not necessary.
-            if (isCalledFromListener) {
-              accountServiceMetrics.accountUpdatesCapturedByScheduledUpdaterCount.inc();
-            }
-            Collection<Account> updatedAccounts = Collections.unmodifiableCollection(idToUpdatedAccounts.values());
-            for (Consumer<Collection<Account>> accountUpdateConsumer : accountUpdateConsumers) {
-              long startTime = System.currentTimeMillis();
-              try {
-                accountUpdateConsumer.accept(updatedAccounts);
-                long consumerExecutionTimeInMs = System.currentTimeMillis() - startTime;
-                logger.trace("Consumer={} has been notified for account change, took {} ms", accountUpdateConsumer,
-                    consumerExecutionTimeInMs);
-                accountServiceMetrics.accountUpdateConsumerTimeInMs.update(consumerExecutionTimeInMs);
-              } catch (Exception e) {
-                logger.error("Exception occurred when notifying accountUpdateConsumer={}", accountUpdateConsumer, e);
-              }
-            }
-          } else {
-            logger.debug("HelixAccountService is updated with 0 updated account");
-          }
+          AccountInfoMap oldAccountInfoMap = accountInfoMapRef.getAndSet(newAccountInfoMap);
+          notifyAccountUpdateConsumers(newAccountInfoMap, oldAccountInfoMap, isCalledFromListener);
         }
       }
     } finally {
       lock.unlock();
     }
   }
 
+  /**
+   * Logs and notifies account update {@link Consumer}s about any new account changes/creations.
+   * @param newAccountInfoMap the new {@link AccountInfoMap} that has been set.
+   * @param oldAccountInfoMap the {@link AccountInfoMap} that was cached before this change.
+   * @param isCalledFromListener {@code true} if the caller is the account update listener, {@@code false} otherwise.
+   */
+  private void notifyAccountUpdateConsumers(AccountInfoMap newAccountInfoMap, AccountInfoMap oldAccountInfoMap,
+      boolean isCalledFromListener) {
+    Map<Short, Account> oldIdToAccountMap = oldAccountInfoMap.idToAccountMap;
+    Map<Short, Account> idToUpdatedAccounts = new HashMap<>();
```

Comment:
looks like this is refactoring again, no change in code, correct?

Reply:
Yes, this was just a refactoring